pkg/lib/template/template_loader.go

```diff
@@ -409,6 +410,30 @@ func (loader *Loader) Render(
 			}
 			return t, nil
 		},
+		"oauth2_client": func(client string) (c *util.OAuthClientConfig, err error) {
+			oAuthClient, ok := loader.OAuthClient[client]
+			if !ok {
+				return nil, errors.Errorf("OAuth client %s not configured", client)
+			}
+			oAuthClient.Key = client
+			return &oAuthClient, nil
+		},
+		"query_escape": func(in string) string {
+			return url.QueryEscape(in)
+		},
+		"query_unescape": func(in string) (string, error) {
+			return url.QueryUnescape(in)
+		},
+		"base64_encode": func(in string) string {
+			return base64.StdEncoding.EncodeToString([]byte(in))
+		},
+		"base64_decode": func(in string) (string, error) {
+			b, err := base64.StdEncoding.DecodeString(in)
+			if err != nil {
+				return "", err
+			}
+			return string(b), err
```

Comment:
The problem with returning errors in template functions is that the template stops rendering. I think in our case its better to return err.Error() as the value